addons/website/models/res_lang.py

```diff
@@ -19,8 +18,6 @@ def write(self, vals):
     @api.model
     @tools.ormcache_context(keys=("website_id",))
     def get_available(self):
-        website = ir_http.get_request_website()
-        if not website:
-            return super().get_available()
-        # Return the website-available ones in this case
-        return request.website.language_ids.get_sorted()
+        if request and getattr(request, 'is_frontend', True):
+            return self.env['website'].get_current_website().language_ids.get_sorted()
```

Comment:
get_current_website gets the website attached to the request object. With my work it is attached later (via `_frontend_pre_dispatch()`) but we whatever is lang-related needs to work before saving website on the request object because `_match()` does a lot lang-related. Using `get_current_website()` the current website is saved in the env cache and later reused in `_frontend_pre_dispatch()` to set the `request.website` attribute (that is used in qweb templates)